x-pack/plugins/cases/common/registry.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+interface BaseItem {
+  id: string;
+}
+
+export class CaseRegistry<T extends BaseItem> {
```

Comment:
Same code as in x-pack/plugins/cases/public/client/attachment_framework/registry.ts. The class name changed CaseRegistry because both the front end and the backend can use it.